eng/tools/mcp/azure-sdk-java-mcp/src/index.ts

```diff
@@ -0,0 +1,282 @@
+                    "The local absolute path to the tspconfig.yaml file. Example: C:\\workspace\\azure-rest-api-specs\\specification\\communication\\Communication.Messages\\tspconfig.yaml"
+                ),
+            remoteTspConfigUrl: z
+                .string()
+                .optional()
+                .describe(
+                    "The remote URL to the tspconfig.yaml file. The URL must contain a commit id, not a branch name. Example: https://github.com/Azure/azure-rest-api-specs/blob/dee71463cbde1d416c47cf544e34f7966a94ddcb/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml"
+                ),
+            },
+    annotations: {
+      title: "Sync Java SDK",
+    },
+   },
+    async (args) => {
+        logToolCall("sync_java_sdk");
+        const result = await initJavaSdk(
+            args.localTspConfigPath,
+            args.remoteTspConfigUrl,
+        );
+        return result;
+    },
+);
+
+// Tool: generate_java_sdk
+server.registerTool(
+    "generate_java_sdk",
+    {
+        description:
+            "Generate the Java SDK from TypeSpec source files located in the 'TempTypeSpecFiles' directory within the specified working directory. If 'TempTypeSpecFiles' is not present, prompt the user to specify whether to generate from a local or remote TypeSpec source, and use the sync_java_sdk tool as needed before proceeding.",
+        inputSchema: {
+            cwd: z
+                .string()
+                .describe(
+                    "The absolute path to the working directory containing the 'TempTypeSpecFiles' directory with TypeSpec source files. Example: C:\\workspace\\azure-sdk-for-java\\sdk\\communication\\azure-communication-messages"
+                ),
+        },
+        annotations: {
+            title: "Generate Java SDK",
+        },
+    },
+    async (args) => {
+        logToolCall("generate_java_sdk");
+        const result = await generateJavaSdk(
+            args.cwd,
+            true
+        );
+        return result;
+    },
+);
+
+// Tool: update_client_name
+server.registerTool(
+    "update_client_name",
+    {
+        description:
+      "Update client name for both client.tsp and the generated java sdk. Follow the returned instruction to update old client name to new client name, be sure to ask for old client name and new client name. e.g. MediaMessageContent.mediaUri to MediaMessageContent.mediaUrl",
+    inputSchema: {},
+        annotations: {
+            title: "Update Client Name",
+        },
+    },
+  async () => {
+        logToolCall("update_client_name");
+    const result = await clientNameUpdateCookbook();
+        return result;
+    },
+);
+
+// Tool: prepare_java_sdk_environment
+server.registerTool(
+    "prepare_java_sdk_environment",
+    {
+        description:
+            "Prepare the development environment required for Java SDK generation. This includes setting up Node.js/npm, Java/Maven, and TypeSpec tools.",
```

Comment:
Should you give more detailed discussion about when `prepare_java_sdk_environment` need to be called. E.g. "call this when you see failure on maven compile"

Reply:
Yeah, I also prefer not have it in other tools (as when you have more tool, you duplicate it more).

But the instruction is at another place. E.g. if I only module folder, it won't read that.

Reply:
Do you have suggestion on where to put?

Reply:
Just this description? Agent read description of the tool.

Reply:
Got you, will add something into prepare env's description and see the behavior. And I also want to make sure if user explicit input 'prepare environment', the tool will be executed.

Reply:
I'm doing the checking first if you see the cookbook.

Reply:
I might have follow up pr to address the environment preparation comments and merge this PR first if it is fine for you.

Reply:
I mean, code it in tool (`spawnAsync("java", "-version")` and check output), instead of write it in prompt.

Reply:
Yeah, fine with the PR.